src/options/SMTConfig.h

```diff
@@ -319,6 +320,12 @@ struct SMTConfig
   vec<char*>    info_names;
   Map<const char*,Info*,StringHash,Equal<const char*> >   infoTable;
   Map<const char*,SMTOption*,StringHash,Equal<const char*> > optionTable;
+
+  bool usedForInitialization = false; // Some options can be changed only before this config is used for initialization of MainSolver
+  bool isPreInitializationOption(const char* o_name) {
+      return strcmp(o_name, o_produce_inter) == 0 || strcmp(o_name, o_produce_proofs) == 0;
```

Comment:
We should add:
 - o_sat_pure_lookahead
 - o_sat_lookahead_split
 - o_ghost_vars

Reply:
These options have been added.